src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -111,6 +111,9 @@ def get_argument_min_nodes(self):
     def get_argument_max_nodes(self):
         return self.get_param("max_nodes")
 
+    def get_argument_infrastructure_resource_group(self):
```

Comment:
Please move logic for new argument to `ContainerappEnvPreviewCreateDecorator`.
And add logic in `validate_arguments()` and `construct_payload()`.
For example:  `ContainerAppPreviewCreateDecorator`

Here is a sample:

class ContainerappEnvPreviewCreateDecorator(ContainerappEnvCreateDecorator):

    def construct_payload(self):
        super().construct_payload()
        self.set_up_infrastructure_resource_group()

    def validate_arguments(self):
        super().validate_arguments()
        # logic for validate argument infrastructure_resource_group

Reply:
Moved